src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/overrides/ConfigOverridesApplier.java

```diff
@@ -0,0 +1,127 @@
+      return;
+    }
+
+    try {
+      long lastUpdatedTimestamp = Long.parseLong(lastUpdatedTimestampString);
+      LOG.debug("Last updated(writer): {}, Last applied(reader): {}", lastUpdatedTimestamp,
+          lastAppliedTimestamp);
+      if (lastUpdatedTimestamp > lastAppliedTimestamp) {
+        apply(ConfigOverridesHelper.deserialize(overridesJson));
+      } else {
+        LOG.debug("Not applying override. Last updated timestamp {} is behind last applied "
+            + "timestamp {}", lastUpdatedTimestamp, lastAppliedTimestamp);
+      }
+    } catch (IOException ioe) {
+      LOG.error("Unable to deserialize overrides JSON:" + overridesJson, ioe);
+    } catch (NumberFormatException nfe) {
+      LOG.error("Unable to parse the lastUpdatedTimestamp {} string as a number.",
+          lastUpdatedTimestampString, nfe);
+    }
+  }
+
+  private void apply(final ConfigOverrides overrides) {
+    if (PerformanceAnalyzerApp.getRcaController() != null && PerformanceAnalyzerApp.getRcaController().isRcaEnabled()) {
+      LOG.info("Applying overrides: {}", overrides.getEnable().getRcas());
+      RcaConf rcaConf = PerformanceAnalyzerApp.getRcaController().getRcaConf();
+      if (rcaConf != null) {
+        Set<String> currentMutedRcaSet = new HashSet<>(rcaConf.getMutedRcaList());
+        Set<String> currentMutedDeciderSet = new HashSet<>(rcaConf.getMutedDeciderList());
+        Set<String> currentMutedActionSet = new HashSet<>(rcaConf.getMutedActionList());
+        // check and remove any nodes that are in the disabled list that were enabled just now.
+        if (overrides.getEnable() != null) {
+          if (overrides.getEnable().getRcas() != null) {
+            currentMutedRcaSet.removeAll(overrides.getEnable().getRcas());
+          }
+          if (overrides.getEnable().getDeciders() != null) {
+            currentMutedDeciderSet.removeAll(overrides.getEnable().getDeciders());
+          }
+          if (overrides.getEnable().getActions() != null) {
+            currentMutedActionSet.removeAll(overrides.getEnable().getActions());
+          }
+        }
+
+        // union the remaining already disabled nodes with the new set of disabled nodes.
+        if (overrides.getDisable() != null) {
+          if (overrides.getDisable().getRcas() != null) {
+            currentMutedRcaSet.addAll(overrides.getDisable().getRcas());
+          }
+          if (overrides.getDisable().getDeciders() != null) {
+            currentMutedDeciderSet.addAll(overrides.getDisable().getDeciders());
+          }
+          if (overrides.getDisable().getActions() != null) {
+            currentMutedActionSet.addAll(overrides.getDisable().getActions());
+          }
+        }
+
+        LOG.debug("New set of muted rcas: {}", currentMutedRcaSet);
+        LOG.debug("New set of muted deciders: {}", currentMutedDeciderSet);
+        LOG.debug("New set of muted actions: {}", currentMutedActionSet);
```

Comment:
We should be applying change when someone calls the API or at startup, which shouldn't be that many. Should we make it info logs ? Or add a metric.